### DISCUSSION TITLE
Contribution to Creating Shareable Albums from Niffler Datasets

### DISCUSSION THREAD

**Discussion by PineappleTech1213** [Emory-HITI/Niffler#397] [Ideas]
Hello everyone, this is Zheng Zeng a newly graduate from Carnegie Mellon University. I major in Master of Information System Management and I intend to pursue a PhD degree in Bioinformatics, and I believe this project can be a good start for me. After going through the project introduction, I have a few ideas towards how to integrate Kheops to create shareable albums of Niffler datasets.

Goal: to create shareable albums with Niffler datasets from MongoDB on Kheops.

Implementations:
1. Build an API similar to DICOMWeb API for retrieving data from Niffler datasets
2. Use Kheops to create an album object and set it to be the designated location to store the retrieved data
3. Deploy the album on cloud with a link for easy access with access codes for users if necessary.

I have seen similar proposals in the discussion panel, and I am hoping to learn more about it with other developers! Thank you for your response and ideas!

**Comment by pradeeban** [comment]
Hello and welcome!

Your suggestions are good.

> Deploy the album on cloud with a link for easy access with access codes for users if necessary.

We most likely will need to avoid this. The images can contain PHI (not necessarily fully anonymized). So the data likely remains internally in local storage. The albums are meant to be shared inside a healthcare organization, and deployed in a server or a cluster rather than through a cloud.

**Reply by PineappleTech1213** [reply]
Thank you! Good to know that. Then it should be only accessible within the organization domain.

**Reply by pradeeban** [reply]
Yes, that is the expected feature and default deployment need. 

From the coding standpoint, whether you host it in a public cloud or a private cloud/cluster makes no difference. It is just a matter of practical needs and limitations.
